Sources/StreamChatUI/Utils/LoaderView.swift

```diff
@@ -0,0 +1,307 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+extension UIView {
+    func showLoader(heightToCornerRadiusRatio: CGFloat) {
+        isUserInteractionEnabled = false
+        if self is UITableView {
+            ListLoader.addLoaderTo(self as! UITableView, heightToCornerRadiusRatio: heightToCornerRadiusRatio)
+        } else if self is UICollectionView {
+            ListLoader.addLoaderTo(self as! UICollectionView, heightToCornerRadiusRatio: heightToCornerRadiusRatio)
+        } else {
+            ListLoader.addLoaderToViews([self], heightToCornerRadiusRatio: heightToCornerRadiusRatio)
+        }
+    }
+    
+    func hideLoader() {
+        isUserInteractionEnabled = true
+        if self is UITableView {
+            ListLoader.removeLoaderFrom(self as! UITableView)
+        } else if self is UICollectionView {
+            ListLoader.removeLoaderFrom(self as! UICollectionView)
+        } else {
+            ListLoader.removeLoaderFromViews([self])
+        }
+    }
+}
+
+protocol ListLoadable {
+    func ld_visibleContentViews() -> [UIView]
+}
+
+extension UITableView: ListLoadable {
+    func ld_visibleContentViews() -> [UIView] {
+        (visibleCells as NSArray).value(forKey: "contentView") as! [UIView]
+    }
+}
+
+extension UICollectionView: ListLoadable {
+    func ld_visibleContentViews() -> [UIView] {
+        (visibleCells as NSArray).value(forKey: "contentView") as! [UIView]
+    }
+}
+
+extension UIColor {
+    static func backgroundFadedGrey() -> UIColor {
+        UIColor(red: (246.0 / 255.0), green: (247.0 / 255.0), blue: (248.0 / 255.0), alpha: 1)
+    }
+    
+    static func gradientFirstStop() -> UIColor {
+        UIColor(red: (238.0 / 255.0), green: (238.0 / 255.0), blue: (238.0 / 255.0), alpha: 1.0)
+    }
+    
+    static func gradientSecondStop() -> UIColor {
+        UIColor(red: (221.0 / 255.0), green: (221.0 / 255.0), blue: (221.0 / 255.0), alpha: 1.0)
+    }
+}
+
+extension UIView {
+    func boundInside(_ superView: UIView) {
+        translatesAutoresizingMaskIntoConstraints = false
+        superView.addConstraints(
+            NSLayoutConstraint.constraints(
+                withVisualFormat: "H:|-0-[subview]-0-|",
+                options: NSLayoutConstraint.FormatOptions(),
+                metrics: nil,
+                views: ["subview": self]
+            )
+        )
+        superView.addConstraints(
+            NSLayoutConstraint.constraints(
+                withVisualFormat: "V:|-0-[subview]-0-|",
```

Comment:
I would recommend to do no use visual constraints, if already using programmatical layout approach, 
this is mixing approaches and making code harder to read